framework/base/Event.php

```diff
@@ -145,11 +143,18 @@ public static function hasHandlers($class, $name)
         } else {
             $class = ltrim($class, '\\');
         }
-        do {
+
+        $classes = array_merge(
```

Comment:
We need to add update PHPDoc for `hasHandlers` method and mention new feature there